java-frontend/src/main/java/org/sonar/java/se/xproc/MethodBehavior.java

```diff
@@ -93,15 +106,34 @@ public void createYield(ExplodedGraph.Node node, boolean storeNodeForReporting)
 
   private void addParameterConstraints(ExplodedGraph.Node node, MethodYield yield) {
     // add the constraints on all the parameters
+    int index = 0;
     for (SymbolicValue parameter : parameters) {
       ConstraintsByDomain constraints = node.programState.getConstraints(parameter);
       if (constraints == null) {
         constraints = ConstraintsByDomain.empty();
+      } else {
+        //cleanup based on signature
+        org.objectweb.asm.Type[] argumentTypes = org.objectweb.asm.Type.getArgumentTypes(signature.substring(signature.indexOf('(')));
+        constraints = cleanup(constraints, argumentTypes[index]);
       }
       yield.parametersConstraints.add(constraints);
+      index++;
     }
   }
 
+  private static ConstraintsByDomain cleanup(ConstraintsByDomain constraints, org.objectweb.asm.Type argumentType) {
+    if (constraints == null || constraints.isEmpty()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "false" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2589)